grade-scripts/shared/utils.js

```diff
@@ -13,6 +15,53 @@ const responseIsOK = (response, siteToCheck) => {
         firstStatusDigit === 3
 }
 
+const getSiteData = (inputPath, fileForSubset) => {
+    // get initial file data
+    let siteDataFiles = fs.readdirSync(inputPath)
+
+    // if we've defined a file for subset, get all the sites listed in that
+    // and avoid processing all the rest
+    if (fileForSubset) {
+        let sitesForSubset = fs.readFileSync(fileForSubset, { encoding: 'utf8' })
+            .trim()
+            .split('\n')
+
+        // we map from the sitesForSubset array because we want
+        // to preserve the order defined in that file
+        siteDataFiles = sitesForSubset
+            .map(hostname => hostname + '.json')
+            .filter((fileName) => {
+                return siteDataFiles.indexOf(fileName) > -1
+            })
+    }
+
+    let siteData = siteDataFiles.map(fileName => require(`${process.cwd()}/${inputPath}/${fileName}`))
+
+    // don't process files that failed in the previous step
+    siteData = siteData.filter(data => !siteData.failed)
+
+    return siteData
+}
+
+const dataFileExists = (hostname, outputPath) => {
+    let destPath = `${outputPath}/${hostname}.json`
+    let fileExists
+
+    try {
+        fileExists = fs.existsSync(destPath)
+    } catch (e) {
+        // ¯\_(ツ)_/¯
```

Comment:
Not sure if this is intentional, but maybe even a console.log() might help identifying the error better here, what do you think?

Reply:
Huh, for some reason I assumed `fs.existsSync` threw an error if the file didn't exist (hence the empty catch). I'll fix :+1: